front/pages/[user]/ds/[name]/search.tsx

```diff
@@ -0,0 +1,303 @@
+                                      </div>
+                                      <div
+                                        className="ml-2 mr-4 flex-1 cursor-pointer border-l-4 border-slate-400"
+                                        onClick={() => {
+                                          expandedChunkId == chunkId
+                                            ? setExpandedChunkId(null)
+                                            : setExpandedChunkId(chunkId);
+                                        }}
+                                      >
+                                        <p
+                                          className={`break-words pl-1 text-xs italic text-gray-500 ${
+                                            expandedChunkId === chunkId
+                                              ? ""
+                                              : "line-clamp-2"
+                                          } `}
+                                        >
+                                          {chunk.text}
+                                        </p>
+                                      </div>
+                                    </div>
+                                  </div>
+                                </div>
+                              </div>
+                            );
+                          })}
+                        </div>
+                      </div>
+                    </li>
+                  ))}
+                <div className="mt-10 flex flex-col items-center justify-center text-sm text-gray-500">
+                  {(() => {
+                    if (error) {
+                      return (
+                        <p className="text-sm font-bold text-red-400">
+                          Something went wrong...
+                        </p>
+                      );
+                    }
+                    if (isLoading) {
+                      return <p>Searching...</p>;
+                    }
+                    if (
+                      !isLoading &&
+                      searchQuery.length == 0 &&
+                      documents.length === 0
+                    ) {
+                      return <p>Please enter your search query.</p>;
+                    }
+                    if (
+                      !isLoading &&
+                      searchQuery.length > 0 &&
+                      documents.length === 0
+                    ) {
+                      return <p>No document found.</p>;
+                    }
+
+                    return <></>;
+                  })()}
+                </div>
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
+    </AppLayout>
+  );
+}
+
+export async function getServerSideProps(context: any) {
+  const authRes = await auth_user(context.req, context.res);
+  if (authRes.isErr()) {
+    return { noFound: true };
+  }
+  const auth = authRes.value();
+
+  const readOnly =
```

Comment:
This page won't work for an anonymous user right, as the providers won't be loaded. Given that you don't display errors returned, we probably want to hide than panel for anonymous users and redirect here if anonymous?

Reply:
Where do you want to redirect the user?

Reply:
If the panel is properly hidden to anonymous user they should never land here. So you can simply redirect to the dataSource page itself or even `/`

Example: https://github.com/dust-tt/dust/blob/main/front/pages/%5Buser%5D/ds/%5Bname%5D/settings.jsx#L261-L268
You can then use that: https://github.com/dust-tt/dust/blob/main/front/pages/%5Buser%5D/ds/%5Bname%5D/settings.jsx#L301